Basic-Car-Maintenance/Shared/Dashboard/Views/DashboardView.swift

```diff
@@ -47,14 +47,24 @@ struct DashboardView: View {
                 }
             }
             .navigationTitle(Text("Dashboard"))
-            .sheet(isPresented: $isShowingAddView) {
+            .sheet(isPresented: $isShowingAddView, onDismiss: {
+                if viewModel.errorOccurred {
+                    viewModel.isShowingErrorAlert = true
+                }
+            }, content: {
```

Comment:
I understand why, but I don't really like this as a solution, because now you have to remember that `errorOccurred` is essentially the same as `isShowingErrorAlert` and that will be forgotten

Can you try this [solution](https://stackoverflow.com/a/75795992/17309983)?
Or if that doesn't work, I'm fine with instead making the `AddMaintenanceView` be pushed onto the `DashboardView` so that you can then present an alert

Thanks for your work on this!

Also note I prefer multiple trailing closure syntax, so the alert modifier would look like this instead:

```swift
.alert("An Error Occurred", isPresented: $viewModel.isShowingErrorAlert) {
    Button("OK", role: .cancel) {
        viewModel.errorOccurred = false
    }
} message: {
    Text("\(viewModel.errorMessage)")
}
```

Reply:
Thanks for suggesting that solution! Unfortunately it only partially worked. Here's what would happen when attaching .alert() to the AddMaintenanceView inside of the .sheet() :

- Go to AddMaintenanceView from Dashboard.
- Press "Add", test error URLError is thrown.
- Sheet dismisses but the alert is not shown.
- Go back to the AddMaintenanceView.
- Alert is now shown because isShowingErrorAlert was set to true when the error was thrown before.

<img width="200" alt="Screenshot 2023-10-02 at 4 52 38 PM" src="https://github.com/mikaelacaron/Basic-Car-Maintenance/assets/111900227/cec45a33-5401-4237-a507-37bab63a14d7">

So I went ahead with pushing AddMaintenanceView to the NavigationStack instead. Using .navigationDestination(isPresented: destination:) allows me to change less of the code, but the documentation says we should prefer binding NavStack to a Path. That would be more scalable, but we are only pushing a single view so maybe it's not necessary. Let me know if you want that instead!

- Giorgio